main/core.js

```diff
@@ -0,0 +1,174 @@
+'use strict'
+
+const { app, BrowserWindow, dialog } = require('electron')
+const { join } = require('node:path')
+const execa = require('execa')
+const wallet = require('./wallet')
+const assert = require('node:assert')
+const fs = require('node:fs/promises')
+const JSONStream = require('jsonstream')
+const Sentry = require('@sentry/node')
+
+/** @typedef {import('./typings').Context} Context */
+/** @typedef {import('./typings').CoreEvent} CoreEvent */
+/** @typedef {import('./typings').Activity} Activity */
+
+const corePath = join(
+  __dirname,
+  '..',
+  'node_modules',
+  '@filecoin-station',
+  'core',
+  'bin',
+  'station.js'
+)
+console.log('Core binary: %s', corePath)
+
+let online = false
```

Comment:
We could remove a bunch of state and methods that was used (or unused) in the Saturn implementation:

- `isReady` this was never checked
- `isRunning` only one place calls `start()` now, once the app starts, so no need to check if there already is a child process. Anyway, `core` would fail to launch twice, due to its lock file. Also, this was never checked from outside.
- `close()` this was never called
- `childLog` `station logs` does this
- `getWebUrl()`

Reply:
I don't remember enough details to be able to help you make a better decision.

Maybe you could land this PR first, so we have something that works, then refactor the code to remove the state-related functionality we no longer need.

Reply:
Ah sorry I misspoke, I was able to remove those functions already, as they are no longer used.